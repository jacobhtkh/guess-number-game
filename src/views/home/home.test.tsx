import { render, screen } from '@testing-library/react';
import { HomeView } from '..';

describe('HomeView', () => {
  describe('@render', () => {
    it('renders without crashing', () => {
      render(<HomeView />);
      const homeView = screen.getByTestId('home-view');

      expect(homeView).toBeInTheDocument();
      expect(homeView).toMatchSnapshot();
    });
  });
});
