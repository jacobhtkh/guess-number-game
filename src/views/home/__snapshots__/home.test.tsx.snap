// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView @render renders without crashing 1`] = `
<div
  class="h-screen flex items-center justify-center"
  data-qa="home-view"
>
  <div
    class="max-w-md border border-solid rounded-md border-sky-600 shadow-md shadow-sky-600"
    data-qa="guess-number-card"
  >
    <div
      class="px-6 py-4 text-center"
    >
      <div
        class="mb-5"
      >
        <h2
          class="mb-3 text-xl font-semibold tracking-tight text-sky-600"
        >
          Guess My Number!
        </h2>
        <h4
          class="mb-2 text-sky-900"
        >
          Think of a number between 1 and 10000.
        </h4>
        <p>
          Click the button below to get the computer to start guessing what it is.
        </p>
      </div>
      <button
        class="px-4 py-2 text-sm shadow bg-sky-100 shadow-sky-600 text-sky-700 hover:bg-sky-600 hover:text-sky-100 rounded-md"
        data-qa="primary-button"
      >
        Start guessing
      </button>
    </div>
  </div>
</div>
`;
